src/use_file.rs

```diff
@@ -68,30 +68,21 @@ fn get_rng_fd() -> Result<libc::c_int, Error> {
 // Succeeds once /dev/urandom is safe to read from
 #[cfg(any(target_os = "android", target_os = "linux"))]
 fn wait_until_rng_ready() -> Result<(), Error> {
-    // Poll /dev/random to make sure it is ok to read from /dev/urandom.
+    // Read a byte from /dev/random to make sure it is ok to read from
+    // /dev/urandom. reading a byte instead of polling is more compatible with
+    // sandboxes that disallow `poll()` but which allow reading /dev/random,
```

Comment:
I think this _sounds_ reasonable, but do we have any documentation or examples of sandboxes which have this issue? It's not obvious to me that `read`-ing from an open file descriptor would be better or worse than `poll`-ing or `select`-ing it. 

The reason for the current implementation (discussed in #58 and https://github.com/briansmith/ring/issues/558#issuecomment-502431497) is to avoid "debiting" a single byte from the `/dev/random` "entropy estimate". Both libsodium and openssl try to avoid this read:
  - [`libsodium` always uses `poll`](https://github.com/jedisct1/libsodium/blob/4a15ab7cd0a4b78a7356e5f488d5345b8d314549/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c#L168)
  - [`openssl` uses `select()` by default but can be configured to use `read()`](https://github.com/openssl/openssl/blob/4cb31128b5790819dfeea2739fbde265f71a10a2/providers/implementations/rands/seeding/rand_unix.c#L468-L484)

Personally, I think the downside of reading one byte from the `/dev/random` pool is not that bad, and I prefer the simplicity of this approch. I just want to understand why `select`/`poll` is not a good fit in practice.

Reply:
>  I just want to understand why select/poll is not a good fit in practice.

I am submitting this change is because I have to write documentation about how to configure sandboxes and when one can avoid doing pre-sandbox initializations. In investigating this, I found that it is actually really difficult to document the seccomp policy for `poll`. The Chromium sandbox has conditional logic for __NR_POLL vs __NR_PPOLL for example. I added some more commentary about this in the comments of this PR.

I am also trying to ensure that if somebody is already using OpenSSL (or a fork of it) then switching to a `getrandom`-based library will not affect their sandbox policies.

So, in some sense it is hypothetical, but that's mostly because I write libraries for orgs who make frameworks for people who product applications.

Reply:
> [openssl uses select() by default but can be configured to use read()](https://github.com/openssl/openssl/blob/4cb31128b5790819dfeea2739fbde265f71a10a2/providers/implementations/rands/seeding/rand_unix.c#L468-L484)

OpenSSL uses `select()` unless there are too many file descriptors open, in which case it will use `read()`.

Reply:
Those additional comments look great! Thanks for the explanation, makes sense to me!